collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceGoogleMapActivity.java

```diff
@@ -378,6 +378,14 @@ protected void onStop() {
         disableMyLocation();
     }
 
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+
+        if (schedulerHandler != null && !schedulerHandler.isCancelled()) {
```

Comment:
As I said in the description the crash occurs only when we use OSM but we should cancel schedulerHandler here as well. I debugged the project and despite the fact it works without any crash recording location is running after we close the map view.